Bottom: e7a1441856fbd796b95f906b4b4a0bc6e4397335
Top:    ef3d546943241b187eeedbe34dca9af72eac55ab
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-13 01:25:52 +0000

Refresh of hid-ps-add-support-for-properl

---

diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 09db3b2..9080787 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1244,7 +1244,7 @@ ps_fill_polygon (hidGC gc, int n_coords, Coord *x, Coord *y)
 }
 
 static void
-draw_polyarea (hidGC gc, POLYAREA * pa, const BoxType * clip_box)
+fill_polyarea (hidGC gc, POLYAREA * pa, const BoxType * clip_box)
 {
   /* Ignore clip_box, just draw everything */
 
@@ -1275,15 +1275,13 @@ draw_polyarea (hidGC gc, POLYAREA * pa, const BoxType * clip_box)
 static void
 ps_draw_pcb_polygon (hidGC gc, PolygonType * poly, const BoxType * clip_box)
 {
-  use_gc (gc);
-
-  draw_polyarea (gc, poly->Clipped, clip_box);
+  fill_polyarea (gc, poly->Clipped, clip_box);
   if (TEST_FLAG (FULLPOLYFLAG, poly))
     {
       POLYAREA *pa;
 
       for (pa = poly->Clipped->f; pa != poly->Clipped; pa = pa->f)
-        fill_polyarea (pa, clip_box, scale);
+        fill_polyarea (gc, pa, clip_box);
     }
 }
